File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.py
from math import sqrt
def solution(n, k):
    answer = 0
    # k 진수로 바꾸고 이걸 문자열로 바꿔서 0을 기준으로 split
    # 그 수들을 int로 형 변환을 해서 소수인지 판별
    
    # 1. n을 k진수로 변환
    num = ''
    while n > 0:
        tmp = str(n % k)
        num = tmp + num
        n //= k
    arr = num.split("0")
    for a in arr:
        if a != '':
            number = int(a)
            if number == 1:
                continue
            print(number)
            prime = True
            for i in range(2, int(sqrt(number))+1):
                if number % i == 0:
                    prime = False
                    break
                    
            if prime:
                answer += 1
    '''
    11 1
    5 1
    2 0
    1
    '''
    return answer

File: 프로그래머스/2/92335. k진수에서 소수 개수 구하기/k진수에서 소수 개수 구하기.java
class Solution {
    public int solution(int n, int k) {
        int answer = 0;
        // 1. n을 k진수로 변환하기
        String kNum = Integer.toString(n, k);
        String[] arr = kNum.split("0");
        
        // 2. 소수 판별
        for (String s : arr) {
            if (s.equals("")) continue;
            
            long num = Long.parseLong(s);
            if (num == 1) continue;
            if (check(num)) answer++;
        }
        
        return answer;
    }
    static boolean check(long num) {
        long limit = (long) Math.sqrt(num);
        for (long i = 2; i <= limit; i++) {
            if (num % i == 0) {
                return false;
            }
        }
        return true;
    }
}